com.unity.ml-agents/Runtime/Inference/BarracudaModelParamLoader.cs

```diff
@@ -406,15 +627,101 @@ static IEnumerable<string> CheckOutputTensorPresence(Model model, int memory)
                 }
 
                 sensorSizes += "]";
-                return $"Vector Observation Size of the model does not match. Was expecting {totalVecObsSizeT} " +
+                return new FailedCheck
+                {
+                    CheckType = CheckType.Warning,
+                    Message =
+                $"Vector Observation Size of the model does not match. Was expecting {totalVecObsSizeT} " +
                     $"but received: \n" +
                     $"Vector observations: {vecObsSizeBp} x {numStackedVector}\n" +
                     $"Total [Observable] attributes: {observableAttributeTotalSize}\n" +
-                    $"Sensor sizes: {sensorSizes}.";
+                    $"Sensor sizes: {sensorSizes}."
+                };
             }
             return null;
         }
 
+
+        /// <summary>
+        /// Generates failed checks that correspond to inputs shapes incompatibilities between
+        /// the model and the BrainParameters.
+        /// </summary>
+        /// <param name="model">
+        /// The Barracuda engine model for loading static parameters
+        /// </param>
+        /// <param name="brainParameters">
+        /// The BrainParameters that are used verify the compatibility with the InferenceEngine
+        /// </param>
+        /// <param name="sensors">Attached sensors</param>
+        /// <param name="observableAttributeTotalSize">Sum of the sizes of all ObservableAttributes.</param>
+        /// <returns>The list the error messages of the checks that failed</returns>
+        static IEnumerable<FailedCheck> CheckInputTensorShape(
```

Comment:
Can you reuse any of the code between this and CheckInputTensorShapeLegacy? I haven't done a side-by-side diff, but looks like it might be cleaner to make legacy an argument instead.

Same comments apply to CheckInputTensorPresence/CheckInputTensorPresenceLegacy

Reply:
I was thinking reusing code, but I think I still prefer having separate methods (even if it duplicates a lot of the code). My thinking is that is will make it easier to deprecate or isolate this code later and make it easier to test as well.